NickelHook.h

```diff
@@ -64,9 +64,10 @@ struct nh_dlsym {
     bool       optional; // default: false - prevents a failure to resolve the sym from being treated as a fatal error, and sets it to NULL instead
 };
 
-// nh_delete_path deletes the provided path if it exists. It performs some checks
+// nh_delete_* deletes the provided path if it exists. It performs some checks
 // to detect and avoid deleting critical system files
-__attribute__((visibility("default"))) void nh_delete_path(const char *path, bool is_dir);
+__attribute__((visibility("default"))) void nh_delete_file(const char *path);
+__attribute__((visibility("default"))) void nh_delete_dir(const char *path);
```

Comment:
Should these return a value to indicate success/error? The code already logs errors internally.

Reply:
I'm not sure. If anything, I think it'd be useful for it to return a bool, but for existing/not existing (afterwards) rather than success/error.

Reply:
My feeling is if we want to return the final state of the file/directory, it's going to require something other than a bool, otherwise there's no way to distinguish errors, unless the function sets an error pointer.

But this all seems to add extra complexity, for what seems to be little gain IMHO.

I've currently got the functions returning true if the file/dir was deleted, or false otherwise. Still open to change.